optuna/trial.py

```diff
@@ -372,14 +372,14 @@ def should_prune(self, step):
 
         Args:
             step:
-                Step of the trial (e.g., epoch of neural network training).
+                Deprecated: Step of the trial (e.g., epoch of neural network training).
 
         Returns:
             A boolean value. If :obj:`True`, the trial should be pruned. Otherwise, the trial will
             be continued.
         """
-
-        # TODO(akiba): remove `step` argument
+        if step is None:
+            step = self.storage.get_latest_step(self._trial_id)
```

Comment:
I think `max(self.storage.get_trial(self._trial_id).intermediate_values.keys())` is better for this purpose because of the following reasons:
- (a) `BaseStorage` already has a lot of abstract methods, so we would like to avoid adding new methods if possible:
  - Of course, by adding `BaseStorage.get_latest_step()`, we can acheive better performance
  - However, I believe here is not a bottleneck point, so this optimization isn't necessary (maybe)
- (b) We're planning to update `BasePruner` interface as well as `BaseSampler`:
  - The new `prune` method will take `FrozenTrial` instead of `trial_id`:
    - It means the latest step of a trial can be easily retrieved from the `FrozenTrial` object
  - `step` argument may be removed from the new `prune` method (for now, this is not sure though)
  - => `get_latest_step` is likely to be unnecessary in the future

Reply:
Thank you for your review. I commit the changes to apply your feedbacks.

> The new prune method will take FrozenTrial instead of trial_id

It sounds good 😄